### PR TITLE
[BUG] - Re: Parting shot debuff applies even if move fails

### LINKED ISSUE #3112: [Bug] Prankster Parting Shot used on Dark types shows "it doesn't affect [target]" but still shows stats lowering
### Describe the bug

As it says in the title. Saw this when my egg move Cottonee was switched into by a trainer with a Poochyena. The switch from Parting Shot doesn't happen, but the stat drops appear to.

### User data export file

_No response_

### Session export file

_No response_

### Expected behavior

All moves that activate Prankster should completely fail when used on Dark types.

### Screenshots / Videos

_No response_

### What OS did you observe the bug on?

PC/Windows

### If other please specify

_No response_

### Which browser do you use?

Firefox

### If other please specify

_No response_

### Additional context

Occurred on locally hosted instance but latest version pulled from Git.

### PR DESCRIPTION
**Pull request by DustinLin**
duplicate of Pr #3135 
---

## What are the changes?

When parting shot fails, the debuffs no longer apply

## Why am I doing these changes?

Closes #3112 

## What did change?

Added check in move workflow that doesn't apply a move's attributes when a move fails

### Screenshots/Videos

- `beta`

https://github.com/user-attachments/assets/d4922730-e09f-4654-9d3e-7ab213b9a73f

https://github.com/user-attachments/assets/837e7ae6-c58f-402c-a2aa-0ed25b18578e

- `pr`

https://github.com/user-attachments/assets/05395f98-8365-47b5-81a3-a699db4c8d63

https://github.com/user-attachments/assets/5dc9342b-2747-47b3-a9a6-305e7e6d4518

## How to test the changes?

Using overrides, the 2 cases were:
- user of parting shot has ability Prankster and the opponent is a dark type
- opponent has the ability Good as Gold

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Tempo-anon** [comment]
Hmmm are there any cases where a move fails but something still happens?

**Comment by DustinLin** [comment]
we were discussing this [here](https://github.com/pagefaultgames/pokerogue/pull/3135#issuecomment-2253181505)

and couldn't think of anything. 

But yeah if we could get more thoughts on it that would be good

File: src/phases.ts [hunk 1]
```diff
@@ -2983,7 +2983,7 @@ export class MoveEffectPhase extends PokemonPhase {
           }
 
           applyAttrs.push(new Promise(resolve => {
-            applyFilteredMoveAttrs((attr: MoveAttr) => attr instanceof MoveEffectAttr && attr.trigger === MoveEffectTrigger.PRE_APPLY && (!attr.firstHitOnly || firstHit) && (!attr.lastHitOnly || lastHit),
+            applyFilteredMoveAttrs((attr: MoveAttr) => attr instanceof MoveEffectAttr && attr.trigger === MoveEffectTrigger.PRE_APPLY && (!attr.firstHitOnly || firstHit) && (!attr.lastHitOnly || lastHit) && hitResult !== HitResult.NO_EFFECT,
```
Comment (innerthunder) [inline src/phases.ts:2986]: How does this change affect charge moves (e.g. Solar Beam)?

Comment (DustinLin) [reply src/phases.ts:2986]: Seems fine from testing

Uploading solar_beam.mov…

**Comment by Snailman11** [comment]
This one is likely related: (GaG getting debuffed by PS) https://github.com/pagefaultgames/pokerogue/issues/3283

and link to discord 
https://discord.com/channels/1125469663833370665/1260240283271168152

-Soundproof Parting shot is mentioned in the discord but hasn't yet been mentioned here.

**Pull request merged**